DungeonBuilderC/dungeon_data.cpp

```diff
@@ -61,13 +61,13 @@ string DungeonEntity::getPrimaryName() const
 }
 
 
-string DungeonEntity::vectorStringToJSON(vector<string> *v) const
```

Comment:
Yeah, a few things:
- Pass by reference is generally safer than pointers (would you agree?).
- My compiler tells me that passing the return value of a function into this function as an argument is unsafe enough that it constitutes an error.
- By adding the const attribute to the signature (here, and also in join()) the compiler is happy and the code compiles.

Without the const attribute here I see compiler errors such as: 
`note: candidate function not viable: expects an l-value for argument`

Beyond that, I can't say for sure.